```
opencv> Scanning dependencies of target opencv_xphoto
opencv> [ 70%] Building CXX object modules/xphoto/CMakeFiles/opencv_xphoto.dir/src/bm3d_image_denoising.cpp.o
opencv> [ 70%] Building CXX object modules/xphoto/CMakeFiles/opencv_xphoto.dir/src/inpainting.cpp.o
opencv> [ 70%] Building CXX object modules/xphoto/CMakeFiles/opencv_xphoto.dir/src/dct_image_denoising.cpp.o
opencv> [ 71%] Building CXX object modules/xphoto/CMakeFiles/opencv_xphoto.dir/src/learning_based_color_balance.cpp.o
opencv> [ 71%] Building CXX object modules/xphoto/CMakeFiles/opencv_xphoto.dir/src/grayworld_white_balance.cpp.o
opencv> [ 71%] Building CXX object modules/xphoto/CMakeFiles/opencv_xphoto.dir/src/oilpainting.cpp.o
opencv> [ 71%] Linking CXX executable ../../bin/opencv_annotation
opencv> [ 71%] Building CXX object modules/xphoto/CMakeFiles/opencv_xphoto.dir/src/simple_color_balance.cpp.o
opencv> [ 71%] Building CXX object modules/xphoto/CMakeFiles/opencv_xphoto.dir/src/tonemap.cpp.o
opencv> /nix/store/p1y0xl8dp4s1x1vvxxb5sn84wj6lsh8s-binutils-2.31.1/bin/ld: ../../lib/libopencv_imgcodecs.so.4.3.0: undefined reference to `Imf_opencv::Chromaticities::Chromaticities(Imath_2_3::Vec2<float> const&, Imath_2_3::Vec2<float> const&, Imath_2_3::Vec2<float> const&, Imath_2_3::Vec2<float> const&)'
opencv> /nix/store/p1y0xl8dp4s1x1vvxxb5sn84wj6lsh8s-binutils-2.31.1/bin/ld: ../../lib/libopencv_imgcodecs.so.4.3.0: undefined reference to `Imf_opencv::Header::Header(int, int, float, Imath_2_3::Vec2<float> const&, float, Imf_opencv::LineOrder, Imf_opencv::Compression)'
opencv> collect2: error: ld returned 1 exit status
opencv> make[2]: *** [apps/annotation/CMakeFiles/opencv_annotation.dir/build.make:106: bin/opencv_annotation] Error 1
opencv> make[1]: *** [CMakeFiles/Makefile2:5380: apps/annotation/CMakeFiles/opencv_annotation.dir/all] Error 2
opencv> make[1]: *** Waiting for unfinished jobs....
opencv> [ 71%] Linking CXX executable ../../bin/opencv_visualisation
opencv> /nix/store/p1y0xl8dp4s1x1vvxxb5sn84wj6lsh8s-binutils-2.31.1/bin/ld: ../../lib/libopencv_imgcodecs.so.4.3.0: undefined reference to `Imf_opencv::Chromaticities::Chromaticities(Imath_2_3::Vec2<float> const&, Imath_2_3::Vec2<float> const&, Imath_2_3::Vec2<float> const&, Imath_2_3::Vec2<float> const&)'
opencv> /nix/store/p1y0xl8dp4s1x1vvxxb5sn84wj6lsh8s-binutils-2.31.1/bin/ld: ../../lib/libopencv_imgcodecs.so.4.3.0: undefined reference to `Imf_opencv::Header::Header(int, int, float, Imath_2_3::Vec2<float> const&, float, Imf_opencv::LineOrder, Imf_opencv::Compression)'
opencv> collect2: error: ld returned 1 exit status
opencv> make[2]: *** [apps/visualisation/CMakeFiles/opencv_visualisation.dir/build.make:106: bin/opencv_visualisation] Error 1
opencv> make[1]: *** [CMakeFiles/Makefile2:5411: apps/visualisation/CMakeFiles/opencv_visualisation.dir/all] Error 2
opencv> [ 71%] Linking CXX shared library ../../lib/libopencv_bioinspired.so
opencv> [ 71%] Built target opencv_bioinspired
opencv> [ 71%] Linking CXX shared library ../../lib/libopencv_dnn.so
opencv> [ 71%] Built target opencv_dnn
opencv> [ 71%] Linking CXX shared library ../../lib/libopencv_calib3d.so
opencv> [ 71%] Built target opencv_calib3d
opencv> [ 71%] Linking CXX shared library ../../lib/libopencv_xphoto.so
opencv> [ 71%] Built target opencv_xphoto
opencv> make: *** [Makefile:180: all] Error 2
builder for '/nix/store/yx0hv8h3mjfq19z5prm07l8mfvsbd37c-opencv-4.3.0.drv' failed with exit code 2; last 10 log lines:
  make[1]: *** [CMakeFiles/Makefile2:5411: apps/visualisation/CMakeFiles/opencv_visualisation.dir/all] Error 2
  [ 71%] Linking CXX shared library ../../lib/libopencv_bioinspired.so
  [ 71%] Built target opencv_bioinspired
  [ 71%] Linking CXX shared library ../../lib/libopencv_dnn.so
  [ 71%] Built target opencv_dnn
  [ 71%] Linking CXX shared library ../../lib/libopencv_calib3d.so
  [ 71%] Built target opencv_calib3d
  [ 71%] Linking CXX shared library ../../lib/libopencv_xphoto.so
  [ 71%] Built target opencv_xphoto
  make: *** [Makefile:180: all] Error 2
[0 built (1 failed), 0.0 MiB DL]
error: build of '/nix/store/yx0hv8h3mjfq19z5prm07l8mfvsbd37c-opencv-4.3.0.drv' failed
```

--- a/pkgs/development/libraries/opencv/4.x.nix
+++ b/pkgs/development/libraries/opencv/4.x.nix
@@ -38,20 +38,20 @@
 assert blas.implementation == "openblas" && lapack.implementation == "openblas";
 
 let
-  version = "4.1.2";
+  version = "4.3.0";
 
   src = fetchFromGitHub {
     owner  = "opencv";
     repo   = "opencv";
     rev    = version;
-    sha256 = "0c98ziwvfrzdzwn52a36d37n5rac8zmxq2jn479bzfaii1bib8xx";
+    sha256 = "1r9bq9p1x99g2y8jvj9428sgqvljz75dm5vrfsma7hh5wjhz9775";
   };
 
   contribSrc = fetchFromGitHub {
     owner  = "opencv";
     repo   = "opencv_contrib";
     rev    = version;
-    sha256 = "10ryyxhggin5dk5glf4ycyrfryqf50f4bs10biv6nxlrrinm2di4";
+    sha256 = "068b4f95rlryab3mffxs2w6dnbmbhrnpsdgl007rxk4bwnz29y49";
   };
 
   # Contrib must be built in order to enable Tesseract support:
@@ -62,8 +62,8 @@ let
     src = fetchFromGitHub {
       owner  = "opencv";
       repo   = "opencv_3rdparty";
-      rev    = "32e315a5b106a7b89dbed51c28f8120a48b368b4";
-      sha256 = "19w9f0r16072s59diqxsr5q6nmwyz9gnxjs49nglzhd66p3ddbkp";
+      rev    = "a56b6ac6f030c312b2dce17430eef13aed9af274";
+      sha256 = "1msbkc3zixx61rcg6a04i1bcfhw1phgsrh93glq1n80hgsk3nbjq";
     } + "/ippicv";
     files = let name = platform : "ippicv_2019_${platform}_general_20180723.tgz"; in
       if stdenv.hostPlatform.system == "x86_64-linux" then
@@ -168,7 +168,9 @@ stdenv.mkDerivation {
   # Also, work around https://github.com/NixOS/nixpkgs/issues/26304 with
   # what appears to be some stray headers in dnn/misc/tensorflow
   # in contrib when generating the Python bindings:
-  patches = lib.optional enableCuda ./cuda_opt_flow.patch;
+  patches = [
+    ./cmake-don-t-use-OpenCVFindOpenEXR.patch
+  ] ++ lib.optional enableCuda ./cuda_opt_flow.patch;
   postPatch = ''
     sed -i '/Add these standard paths to the search paths for FIND_LIBRARY/,/^\s*$/{d}' CMakeLists.txt
     sed -i -e 's|if len(decls) == 0:|if len(decls) == 0 or "opencv2/" not in hdr:|' ./modules/python/src2/gen2.py

--- a/pkgs/development/libraries/opencv/cmake-don-t-use-OpenCVFindOpenEXR.patch
+++ b/pkgs/development/libraries/opencv/cmake-don-t-use-OpenCVFindOpenEXR.patch
@@ -1,0 +1,55 @@
+From 6d988c08e852379a163ecd20df8639196d84d014 Mon Sep 17 00:00:00 2001
+From: Bernardo Meurer <meurerbernardo@gmail.com>
+Date: Sun, 26 Apr 2020 14:50:25 -0700
+Subject: [PATCH] cmake: don't use OpenCVFindOpenEXR
+
+Use find_package for this.
+---
+ CMakeLists.txt                  |  2 ++
+ cmake/OpenCVFindLibsGrfmt.cmake | 15 +++------------
+ 2 files changed, 5 insertions(+), 12 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 4c0b3880fc..0360469350 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -14,6 +14,8 @@ FATAL: In-source builds are not allowed.
+ ")
+ endif()
+ 
++# the future!
++include(FindPkgConfig)
+ 
+ include(cmake/OpenCVMinDepVersions.cmake)
+ 
+diff --git a/cmake/OpenCVFindLibsGrfmt.cmake b/cmake/OpenCVFindLibsGrfmt.cmake
+index 0beaf19317..4c5e46e615 100644
+--- a/cmake/OpenCVFindLibsGrfmt.cmake
++++ b/cmake/OpenCVFindLibsGrfmt.cmake
+@@ -227,20 +227,11 @@ endif()
+ # --- OpenEXR (optional) ---
+ if(WITH_OPENEXR)
+   ocv_clear_vars(HAVE_OPENEXR)
+-  if(NOT BUILD_OPENEXR)
+-    include("${OpenCV_SOURCE_DIR}/cmake/OpenCVFindOpenEXR.cmake")
+-  endif()
+-
++  pkg_check_modules(OPENEXR OpenEXR)
+   if(OPENEXR_FOUND)
+     set(HAVE_OPENEXR YES)
+-  else()
+-    ocv_clear_vars(OPENEXR_INCLUDE_PATHS OPENEXR_LIBRARIES OPENEXR_ILMIMF_LIBRARY OPENEXR_VERSION)
+-
+-    set(OPENEXR_LIBRARIES IlmImf)
+-    add_subdirectory("${OpenCV_SOURCE_DIR}/3rdparty/openexr")
+-    if(OPENEXR_VERSION)  # check via TARGET doesn't work
+-      set(HAVE_OPENEXR YES)
+-    endif()
++    set(OPENEXR_INCLUDE_PATHS ${OPENEXR_INCLUDE_DIRS})
++    set(OPENEXR_LIBRARIES ${OPENEXR_LIBRARIES})
+   endif()
+ endif()
+ 
+-- 
+2.26.1
+

